### ISSUE TITLE
GitHub Desktop: Extremely slow after an update

### ISSUE THREAD

**Issue by manas-kulkarni** [desktop/desktop#14950]
### The problem

Everything has become slow. Clicking on file -> Diff takes 5-6 seconds to load (spinner keeps spinning in diff window)
The same goes for all other actions. Have tried different repos from different servers but the same result.
I have tried to uninstall, and delete the cache from AppData but doesn't help.

### Release version

Version 3.0.3 (x64)

### Operating system

Windows 10

### Steps to reproduce the behavior

1. Go to the repo
2. Changelist takes too long to load
3. Click on any file in the channels to view diff (5-6 seconds to load diff)

**Comment by tsvetilian-ty** [comment]
👋 @manas-kulkarni could you upload a log file from GitHub Desktop "Help > Show logs..."?

**Comment by manas-kulkarni** [comment]
[2022-07-12.desktop.production.log](https://github.com/desktop/desktop/files/9089778/2022-07-12.desktop.production.log)

**Comment by sergiou87** [comment]
Hi @manas-kulkarni !

If you updated from 3.0.2 to 3.0.3, I don't think you should experience any changes in performance, at least none of the changes included in 3.0.3 seem like potential reasons for such issue. Have you rebooted your machine since the update? I've seen other users in the past getting performance issues in the app resolved like that 🤞

**Comment by manas-kulkarni** [comment]
Have restarted multiple times. Can it have something to do with NVIDIA driver update?

**Comment by tidy-dev** [comment]
Another thing worth trying is to do a clean install. 

1. Close GitHub Desktop
2. Remove all of the files in `%LOCALAPPDATA%\GitHubDesktop\`
3. Reinstall GitHub Desktop from http://desktop.github.com


Note you will have to add your repositories back and if you have more than one in the same directory, a quick way to do so is to drag and drop the repository folders onto the desktop app.

**Comment by manas-kulkarni** [comment]
I have tried that already.

**Comment by tidy-dev** [comment]
Hmm.. @sakadana  mentioned in another similar issue having a gaming pc. Curious if they also had a NVIDIA driver update. We have heard of electron apps (which Desktop is) being impacted by certain graphic cards tho not in this way (was a crash/non start).

**Comment by manas-kulkarni** [comment]
I have GTX 1080 Driver 516.59

**Comment by tsvetilian-ty** [comment]
I have a device with the same driver, and it works correctly. You could try to add `--disable-gpu --disable-software-rasterizer` as arguments to GitHub Desktop. This should disable any GPU acceleration.

![disable_gpu](https://user-images.githubusercontent.com/9341546/178749640-bc108074-8e15-4840-a98b-8e8575fa2e56.gif)

@manas-kulkarni could you record your screen while using GitHub Desktop?

**Comment by manas-kulkarni** [comment]
Uploading GitHub Desktop 2022-07-13 19-28-31.mp4…

Check the time it takes to load the repo and then the one line diff

**Comment by manas-kulkarni** [comment]
https://user-images.githubusercontent.com/30538315/178752090-e8045988-8dff-4e0e-8c98-5f6d4932d478.mp4

Check the time it takes to load the repo and then the one line diff. Before it used to be instant

**Comment by tsvetilian-ty** [comment]
@manas-kulkarni did you try to disable the GPU?

**Comment by manas-kulkarni** [comment]
Yes. Sorry for not mentioning in the last message. It was ran with the args you provided

**Comment by sergiou87** [comment]
@manas-kulkarni do you have any kind of Antivirus or real-time scanner that could make everything slower? 🤔

**Comment by SmartieCodes** [comment]
I've also been experiencing an extreme slowdown recently on GitHub desktop.

An example would be clicking on the changed files in the list, clicking on one to see what has changed takes 5 seconds showing the swirly before it will show the differences. This used be instant.

I should add, I have a NVIDIA GeForce RTX 3080 Ti Laptop GPU.

**Comment by Gillardo** [comment]
Same issue, tried adding the GPU disable arguments, still no difference.  Have NVIDIA GeForce RTX 3080 as well.

Starting Github desktop takes 5-10 seconds just to load a repo, sync'ing/submitting files again takes 10 seconds +, used to be near instant

**Comment by tsvetilian-ty** [comment]
Do you use any antivirus software? 
Also, could you check the Task Manager (Ctrl+Shift+Esc) to see if there is an abnormal load on the CPU, Memory, Disk, or GPU?

**Comment by manas-kulkarni** [comment]
No anti-virus. CPU load looks normal as well

**Comment by SmartieCodes** [comment]
> Do you use any antivirus software?

I'm using WithSecure, but it's mostly disabled as had been causing issues with other programs like steam cloud before.

> Also, could you check the Task Manager (Ctrl+Shift+Esc) to see if there is an abnormal load on the CPU, Memory, Disk, or GPU?

If I step between the change files repeatedly this is how task manager looks:

![image](https://user-images.githubusercontent.com/6124100/178990751-444f3e72-059c-4986-990e-e7dd6e77092d.png)

When I stop:

![image](https://user-images.githubusercontent.com/6124100/178990944-c33b9b95-3a8e-48a2-bb4d-c37f4675e007.png)

**Comment by sergiou87** [comment]
Thank you @LeahPike , that sounds reasonable. I guess the problem is how long those subprocess take to finish their task.

Is this something new of 3.0.5 or did you notice it too on 3.0.3?

You can also try the latest beta from https://desktop.github.com/beta which includes the latest stable release of Electron (v19) and might make a difference.

**Comment by SmartieCodes** [comment]
> Is this something new of 3.0.5 or did you notice it too on 3.0.3?

I've only updated to 3.0.5 today and had noticed the problem for a few days now.

> You can also try the latest beta from https://desktop.github.com/beta which includes the latest stable release of Electron (v19) and might make a difference.

This version is actually a lot better, still minor lag compared to what I'm used it. It's taking maybe 500ms between viewing changes instead of 5 seconds. Thanks for this, much more workable :)

**Comment by tsvetilian-ty** [comment]
@LeahPike, could you check again with the Task Manager if the beta spawns many processes?

**Comment by SmartieCodes** [comment]
> @LeahPike, could you check again with the Task Manager if the beta spawns many processes?

It is no longer spawning so many processes, usually one, occasionally a second when spamming the changed file list.

**Comment by AnttiFlashnode** [comment]
I also have this issue, it began after updating from 3.0.3 to either .4 or .5 (didn't use it between the .4 to .5 update). Everything that was near-instantaneous now takes seconds, and this quickly adds up. I've updated nothing else and changed nothing.

**Comment by sergiou87** [comment]
Thank you all for your feedback 🙇‍♂️ 

@AnttiFlashnode the only difference between 3.0.3 and .4/.5 is that we upgraded git from 2.35.3 to 2.35.4, and that only includes a security fix, so I'm surprised about the difference in performance (but it's not impossible!).

If you (or anyone else) is willing to try, this is the link to download v3.0.3 and see if the problems exist in it: https://centralreleases.blob.core.windows.net/github-desktop/releases/3.0.3-ecd0e109/GitHubDesktopSetup-x64.exe

**Comment by manas-kulkarni** [comment]
Should also add - Slowdown doesnt happen on macOS, only happening on Windows 10

**Comment by AnttiFlashnode** [comment]
> If you (or anyone else) is willing to try, this is the link to download v3.0.3 and see if the problems exist in it: https://centralreleases.blob.core.windows.net/github-desktop/releases/3.0.3-ecd0e109/GitHubDesktopSetup-x64.exe

Yes, the old version is still as quick as ever! I _was_ wondering if there was a way to downgrade. I guess I'll refrain from updating until someone confirms that a new version doesn't suffer from this.

**Comment by Nico-Connor-Personal** [comment]
The slowdown is also happening on my machines since the latest update,  it's reassuring to hear that the new beta doesn't have the same performance issue for some users.

**Comment by sergiou87** [comment]
I wonder if the problem lies on the new Git version _somehow_. One thing maybe you could try for me:
1. Install v3.0.3 from https://centralreleases.blob.core.windows.net/github-desktop/releases/3.0.3-ecd0e109/GitHubDesktopSetup-x64.exe
2. Let it auto-update to v3.0.5
3. Close GitHub Desktop
4. Replace the contents of C:\Users\sergiou87\AppData\Local\GitHubDesktop\app-3.0.5\resources\app\git with the contents from C:\Users\sergiou87\AppData\Local\GitHubDesktop\app-3.0.3\resources\app\git
⚠️ **IMPORTANT** ⚠️ : this will replace git 2.35.4 with 2.35.3, which contains a security vulnerability that might affect you. Please, keep a backup of the contents of that folder so you can go back.
5. Start the app (now it should be 3.0.5) and see if the performance is as good as it was on 3.0.3.

**Comment by manas-kulkarni** [comment]
I have tried multiple git versions going back to 2.34 but same slowdown

**Comment by sergiou87** [comment]
@manas-kulkarni thank you, but did you follow the steps I mentioned? GitHub Desktop bundles and uses its own git binaries, and I'd like to know if changing those specifically make a difference in performance 🤔

**Comment by manas-kulkarni** [comment]
Yes. I followed your steps and replaced the git inside the GitHub desktop folder. Tried many versions from 2.34.0 to 2.37.2. All had same slowdown

**Comment by sergiou87** [comment]
Thank you!! Then I'm terribly puzzled because that is the only change in 3.0.3 vs 3.0.4/5, just those git binaries 🤯

**Comment by jkoplo** [comment]
Just dropping a note that both a coworker and I experienced this same slowdown on 3.0.5

Switching to the beta cleared everything up.

**Comment by adamvanaken** [comment]
> I have a device with the same driver, and it works correctly. You could try to add `--disable-gpu --disable-software-rasterizer` as arguments to GitHub Desktop. This should disable any GPU acceleration.

FYI, this worked for me -- not sure what changed, but suddenly today version 3.1.8 started really slowing down on my system, and I found this Issue.

**Comment by ghost** [comment]
Anyone got stable solution.....??if yes!! Help me also!!

**Comment by matt-richardson** [comment]
EDIT: Gave the machine a reboot, and it's all back to normal speeds. No idea why it suddenly went on a go-slow 🤷 

~I'm seeing extreme slowness now too, after upgrading to `3.3.11 (arm64)` (from 3.3.10 I think).~
~I'm using an M1 Max mac, running Sonoma `14.3.1`.~

~Looking at the console in the dev tools, it's showing things like:~
```
Executing getAuthorIdentity: git var GIT_AUTHOR_IDENT (took 9.580s)
```
~Running this command with the non bundled git is practically instant.~

~Interestingly, the bundled git is `git version 2.39.3`. My local version is `git version 2.39.3 (Apple Git-146)`, so it's not a difference in git versions.~

~I tried to downgrade, but I'm unsure where to find the binaries.~

**Comment by vixipop** [comment]
been facing this ; tried clean install ; and beta ; it didnt work help pls

**Comment by prma85** [comment]
> been facing this ; tried clean install ; and beta ; it didnt work help pls

Same here. New install and nothing :/

**Comment by LuxyCeremy** [comment]
I just fixed it! It disturbs me for weeks!
Just go to Service and disable "Microsoft PC Manager" related service and uninstall it, then github desktop is smooth as before!

Microsoft PC Manager is a Mcafee like software, it blocks many processes when they tries to read files!
I am adding everything to windows defender's white list, but it didn't work. I never thought that MSPC Manager is the culprit!

**Comment by jason-oc** [comment]
- Don't have/don't see "Microsoft PC Manager" on my system (Windows 11 Pro Version 24H2)
- Reboot didn't fix it
- Updating my git version (2.40 to 2.52) didn't fix it
- Reinstalling GitHub Desktop didn't fix it
- Reinstalling GitHub Desktop after removing everything from `%LOCALAPPDATA%\GitHubDesktop\` didn't fix it
- Adding `--disable-gpu --disable-software-rasterizer` launch options didn't fix it
- Contacted my antivirus admin who added an exclusion for me, didn't fix it

Worth noting that after a reinstall of GitHub Desktop, when I have no repos open in GitHub Desktop and I'm looking at the "Let's Get Started!" page, its response to my interactions with it (clicking around menu items) is instant. It's only when I clone a repo that everything goes to s**t.
